### PR TITLE
Retain all heap dumps per JVM lifecycle

### PR DESCRIPTION
**Pull request by gerashegalov**
Prevent FileAlreadyExistsException from suppressing useful heap dumps.

Contributes to #6661

Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/DeviceMemoryEventHandler.scala [hunk 1]
```diff
@@ -18,6 +18,7 @@ package com.nvidia.spark.rapids
 
 import java.io.File
 import java.lang.management.ManagementFactory
```
Comment (abellina) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/DeviceMemoryEventHandler.scala:20]: file needs copyrights updated.

Comment (gerashegalov) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/DeviceMemoryEventHandler.scala:20]: forgot to set `SPARK_RAPIDS_AUTO_COPYRIGHTER=ON` :)

**Comment by gerashegalov** [comment]
build

**Review by tgravescs — DISMISSED** [review]
seems fine and fairly low risk, but I do question if its needed for 22.10, if we have specific customer or instance we want it, I'm ok with it otherwise we should move to 22.12

**Review by abellina — APPROVED** [review]

**Review by abellina — APPROVED** [review]

**Review by tgravescs — APPROVED** [review]

**Pull request merged**